src/users_invites/users-invites.service.ts

```diff
@@ -0,0 +1,73 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { getConnection, getRepository } from 'typeorm';
+import { UsersInvitesEntity } from './entity/users_invites.entity';
+import { UsersEntity } from '../users/entity/users.entity';
+import { InvitesEntity } from '../invites/entity/invites.entity';
+import { JwtService } from '@nestjs/jwt';
+import { EmailService } from '../email/email.service';
+import { RolesEntity } from '../roles/entity/roles.entity';
+import { UserDataDto } from './dto/userData.dto';
+
+@Injectable()
+export class UsersInvitesService {
+  constructor(
+    private emailService: EmailService,
+    private readonly jwtService: JwtService,
+  ) {}
+
+  async createAndSendInvate(data: UserDataDto): Promise<any> {
+    let message = {};
+    const connection = getConnection();
+    const queryRunner = connection.createQueryRunner();
+
+    await queryRunner.connect();
+    await queryRunner.startTransaction();
+    try {
+      const userInvite = await this.createIvite(data);
+
+      const result = await queryRunner.manager.save(userInvite);
+
+      const { user } = result;
+
+      const token = this.jwtService.sign({
+        email: data.email,
+        userId: user.id,
+      });
+
+      await this.emailService.sendInvite(
+        { name: data.firstName, email: data.email },
+        token,
+      );
+
+      await queryRunner.commitTransaction();
+
+      message = { message: 'SUCCESS' };
+    } catch (err) {
+      await queryRunner.rollbackTransaction();
+      message = { message: 'FAILURE', err };
+    } finally {
+      await queryRunner.release();
+    }
+    return message;
+  }
+
+  private async createIvite(data: UserDataDto): Promise<UsersInvitesEntity> {
+    const user = new UsersEntity();
+    user.firstname = data.firstName;
```

Comment:
all that properties can be set in constructor either some method that will make user